azurerm/resource_arm_sql_server.go

```diff
@@ -202,3 +239,25 @@ func resourceArmSqlServerDelete(d *schema.ResourceData, meta interface{}) error
 
 	return future.WaitForCompletionRef(ctx, client.Client)
 }
+
+func expandAzureRmSqlServerIdentity(d *schema.ResourceData) *sql.ResourceIdentity {
+	identities := d.Get("identity").([]interface{})
+	identity := identities[0].(map[string]interface{})
```

Comment:
this assumes there's one element here which may not necessarily be the case - as such can we return an empty object here to ensure this is disabled:

```suggestion
	if len(identities) == 0 {
		return &sql.ResourceIdentity{}
	}
	identity := identities[0].(map[string]interface{})
```